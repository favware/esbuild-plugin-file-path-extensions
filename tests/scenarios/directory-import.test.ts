import esbuild, { type BuildOptions } from 'esbuild';
import { assertFileContent, buildAbsolutePath, createEsbuildConfig } from '../common/util';

function createConfig(): BuildOptions {
  return createEsbuildConfig({
    platform: 'node',
    format: 'esm',
    outExtension: {
      '.js': '.mjs'
    },
    entryPoints: [buildAbsolutePath('../fixtures/build-in/directory-import.mjs')]
  });
}

describe('Importing directories', () => {
  test('GIVEN a directory import THEN appends "index" to path', async () => {
    const config = createConfig();

    await esbuild.build(config);

    await assertFileContent('../fixtures/build-out/directory-import.mjs');
  });
});
