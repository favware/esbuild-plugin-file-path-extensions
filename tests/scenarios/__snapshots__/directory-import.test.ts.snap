// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Importing directories > GIVEN a directory import THEN appends "index" to path 1`] = `
"// tests/fixtures/build-in/directory-import.mjs
import { foo } from "./directory/index.mjs";
export {
  foo
};
"
`;
